docs/source/schema/custom-scalars.md

```diff
@@ -0,0 +1,202 @@
+
+### `serialize`
+
+The `serialize` method converts the scalar's back-end representation to a JSON-compatible format so Apollo Server can include it in an operation response.
+
+In the example above, the `Date` scalar is represented on the backend by the `Date` JavaScript object. When we send a `Date` scalar in a GraphQL response, we serialize it as the integer value returned by the [`getTime` function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTime) of a JavaScript `Date` object.
+
+> Note that Apollo Client cannot automatically interpret custom scalars ([see issue](https://github.com/apollographql/apollo-feature-requests/issues/2)), so your client must define custom logic to deserialize this value as needed.
+
+### `parseValue`
+
+The `parseValue` method converts the scalar's `serialize`d JSON value to its back-end representation.
+
+Apollo Server calls this method when the scalar appears in an incoming query (such as in a field argument).
+
+### `parseLiteral`
+
+When an incoming query string includes a hard-coded value for the scalar, that value is part of the query document's abstract syntax tree (AST). Apollo Server calls the `parseLiteral` method to convert the value's AST representation (which is always a string) to the JSON-compatible format expected by the `parseValue` method (the example above expects an integer).
+
+## Providing custom scalars to Apollo Server
+
+After you define your `GraphQLScalarType` instance, you include it in the same [resolver map](../data/resolvers/#defining-a-resolver) that contains resolvers for your schema's other types and fields:
+
+```js{21-24}
+const { ApolloServer, gql } = require('apollo-server');
+const { GraphQLScalarType, Kind } = require('graphql');
+
+const typeDefs = gql`
+  scalar Date
+
+  type Event {
+    id: ID!
+    date: Date!
+  }
+
+  type Query {
+    events: [Event!]
+  }
+`;
+
+const dateScalar = new GraphQLScalarType({
+  // See definition above
+});
+
+const resolvers = {
+  Date: dateScalar
+  // ...other resolver definitions...
+};
+
+const server = new ApolloServer({
+  typeDefs,
+  resolvers
+});
+```
+
+## Example: Restricting integers to odd values
+
+In this example, we create a custom scalar called `Odd` that can only contain odd integers:
+
+```js{20-31}
+const { ApolloServer, gql } = require('apollo-server');
+const { GraphQLScalarType } = require('graphql');
+const { Kind } = require('graphql/language');
+
+// Basic schema
+const typeDefs = gql`
+  scalar Odd
+
+  type MyType {
+    oddValue: Odd
+  }
+`;
+
+// Validation function
+function oddValue(value) {
+  return value % 2 === 1 ? value : null;
+}
+
+const resolvers = {
+  Odd: new GraphQLScalarType({
+    name: 'Odd',
+    description: 'Odd custom scalar type',
+    parseValue: oddValue,
+    serialize: oddValue,
+    parseLiteral(ast) {
+      if (ast.kind === Kind.INT) {
```

Comment:
Nice catch on the previously incorrect `Kind.Int` when transposing / recreating this example!

Reply:
Ha I *wish* I could take credit for this but I think it was already correct 😄